src/coreComponents/physicsSolvers/kernels-all.json

```diff
@@ -0,0 +1,512 @@
+      "they are also the names of the CMAKE variables which will be set to one of the values specified",
+      "in the combinations dict just prior to cmake calling configure_file() on the template file"
+    ],
+    "constants" :
+    [
+      [ "key", "value" ],
+      [ "note", "All constants will be set for cmake to configure_file() for every instantiation." ]
+    ],
+    "combinations" :
+    {
+      "vars" : [ "each value in this list will be used to generate kernel template files",
+                 "combinatorally with all other value lists" ]
+    },
+    "explicit" : [ "specify a specific combination of the variables to generate",
+                   "specify by choosing one value from each variable you want generated",
+                   "and concating them with '#' (this can by modified internally at the",
+                   "generation call). Duplicate entries are unified internally" ]
+  },
+
+  "SolidMechanicsKernels": {
+    "vars": [ "SUBREGION_TYPE", "CONSTITUTIVE_TYPE", "FE_TYPE" ],
+    "constants": [
+      [ "ExplicitSmallStrainPolicy", "geos::parallelDevicePolicy< GEOS_BLOCK_SIZE >" ],
+      [ "ExplicitFiniteStrainPolicy", "geos::parallelDevicePolicy< GEOS_BLOCK_SIZE >" ],
+      [ "FixedStressThermoPoromechanicsPolicy", "geos::parallelDevicePolicy< GEOS_BLOCK_SIZE >" ],
+      [ "ImplicitSmallStrainNewmarkPolicy", "geos::parallelDevicePolicy< GEOS_BLOCK_SIZE >" ],
+      [ "ImplicitSmallStrainQuasiStaticPolicy", "geos::parallelDevicePolicy< GEOS_BLOCK_SIZE >" ]
+    ],
+    "combinations": {
+      "SUBREGION_TYPE": [
+        "CellElementSubRegion"
+      ],
+      "CONSTITUTIVE_TYPE": [
+        "DamageSpectral<ElasticIsotropic>",
+        "DamageVolDev<ElasticIsotropic>",
+        "Damage<ElasticIsotropic>",
+        "DuvautLionsSolid<DruckerPrager>",
+        "DuvautLionsSolid<DruckerPragerExtended>",
+        "DuvautLionsSolid<ModifiedCamClay>",
+        "DruckerPragerExtended",
+        "ModifiedCamClay",
+        "DelftEgg",
+        "DruckerPrager",
+        "ElasticIsotropic",
+        "ElasticTransverseIsotropic",
+        "ElasticIsotropicPressureDependent",
+        "ElasticOrthotropic"
+      ],
+      "FE_TYPE": [
+        "H1_Hexahedron_Lagrange1_GaussLegendre2",
+        "H1_Wedge_Lagrange1_Gauss6",
+        "H1_Tetrahedron_Lagrange1_Gauss1",
+        "H1_Pyramid_Lagrange1_Gauss5",
+        "H1_Tetrahedron_VEM_Gauss1",
+        "H1_Prism5_VEM_Gauss1",
+        "H1_Prism6_VEM_Gauss1",
+        "H1_Prism7_VEM_Gauss1",
+        "H1_Prism8_VEM_Gauss1",
+        "H1_Prism9_VEM_Gauss1",
+        "H1_Prism10_VEM_Gauss1",
+        "Q1_Hexahedron_Lagrange_GaussLobatto",
+        "Q2_Hexahedron_Lagrange_GaussLobatto",
+        "Q3_Hexahedron_Lagrange_GaussLobatto",
+        "Q4_Hexahedron_Lagrange_GaussLobatto",
+        "Q5_Hexahedron_Lagrange_GaussLobatto",
+        "H1_Hexahedron_VEM_Gauss1",
+        "H1_Wedge_VEM_Gauss1",
+        "H1_Prism11_VEM_Gauss1"
+      ]
+    },
+    "explicit": []
```

Comment:
@corbett5 Can you put an example here and I will fill out the valid combinations?

Reply:
@rrsettgast this is not how it's done, but I'll probably try and change it to accept this format.

```
"explicit": [
  ["CellElementSubRegion", "DamageSpectral<ElasticIsotropic>", "H1_Hexahedron_Lagrange1_GaussLegendre2"],
  ["CellElementSubRegion", "DamageSpectral<ElasticIsotropic>", "H1_Wedge_Lagrange1_Gauss6"]
]
```

The order of the values should be the same as the order of the `vars`. In this case this is

```
"vars": [ "SUBREGION_TYPE", "CONSTITUTIVE_TYPE", "FE_TYPE" ]
```